### DISCUSSION TITLE
Two issues regarding json::value methods

### DISCUSSION THREAD

**Discussion by vthierry** [nlohmann/json#2792] [Q&A] [answered]
At my application level, it will really help to have the .value() function to 

1/ Apply also on array not only object, e.g., being able to use `my_json_array.value(2, "2")`  on `[0, 1]`.

2/ Enjoying a boolean or numeric to string conversion, e.g., being able to use `my_json_array.value(0, "?")`  on `[0, 1]` returning `"0"`

To this end i very simply define
```
class my_json : public nlohmann::ordered_json {
  // Here value method re-implementation
};
```
but this does NOT work, for instance
```
my_json j = nlohmann::ordered_json::parse("[0,1]");
```
throws 
```
terminate called after throwing an instance of 'nlohmann::detail::type_error'
  what():  [json.exception.type_error.309] cannot use insert() with null
	/usr/include/nlohmann/json.hpp:22030
	/usr/include/nlohmann/json.hpp:22047
	/usr/include/c++/10/bits/stl_iterator.h:886
	/usr/include/c++/10/bits/stl_algo.h:4320 (discriminator 2)
	/usr/include/nlohmann/json.hpp:3626 (discriminator 4)
	/usr/include/nlohmann/json.hpp:3649
	/usr/include/nlohmann/json.hpp:3791
	/usr/include/nlohmann/json.hpp:4387
	/usr/include/nlohmann/json.hpp:19422
	/usr/include/nlohmann/json.hpp:19724
```

advice would be greatly appreciated.

**Comment by nlohmann** [comment, marked as answer]
Sorry for the late response.

1. Such an extension seems doable, but has not yet been requested. I'm hesitant to add it though, because the API is already quite crowded. If you need this, you can implement such a `value` function on top of the public API.
2. This is out of scope of this library. As `value` returns a `basic_json` value, you can call dump for a serialization to string.

Note the class `basic_json` was not designed to be sub-classed.

**Comment by vthierry** [comment]
Dear Niels

     Thanks for your double answer and advices, precious.

Best regards.
